plugins/metrics/pkg/agent/chunks.go

```diff
@@ -0,0 +1,128 @@
+package agent
+
+import (
+	"fmt"
+
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/samber/lo"
+)
+
+type WriteLimit struct {
+	// GrpcMaxBytes is the maximum siz of a message that can be sent over the grpc connection.
+	GrpcMaxBytes int
+
+	// CortexIngestionRateLimit is the maximum number of samples that can be sent to cortex per second.
+	CortexIngestionRateLimit int
+}
+
+func splitNChunks[T any](a []T, n int) ([][]T, error) {
+	if n == 0 {
```

Comment:
What happens if `n == 1`?

Reply:
when `n == 1` the populating loop below will iterate once, then `end` will exceed the length of the given slice, and finish. `splitNChunks([]int{0, 1, 2, 3}, 1) == [][]int{{0, 1, 2, 3}}`